### PR TITLE
MAUI - 870300 - Prepare the UG documentation for the unbound column support 

### PR DESCRIPTION
**Pull request by AbhishekSF4201**
Added UG document for MAUI Data Grid Unbound column.
Added Sort icon color and Data grid lines

### CODE REVIEW

File: MAUI/DataGrid/Unbound-Column.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    {
+        if (e.UnboundAction == DataGridUnboundActions.QueryData)
+            {
+                var Price1 = Convert.ToInt16(e.Record!.GetType().GetProperty("Price1")?.GetValue(e.Record));
+                var Price2 = Convert.ToInt16(e.Record!.GetType().GetProperty("Price2")?.GetValue(e.Record));
+                var amount = Price1 * Price2;
+                e.Value = amount;
+            }
+    }
+{% endhighlight %}
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Unbound-Column.md:165]: Removed the C# code

File: MAUI/DataGrid/Unbound-Column.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+{% endhighlight %}
+{% endtabs %}
+
+
+<img alt="" src="Images\unbound-column\maui-datagrid-sample-example.png" width="404" Height = "396" />
+
+N> It is mandatory to specify the GridColumn.MappingName for SfDataGrid.GridUnboundColumn with some name to identify the column. It is not necessary to define name of the field in the data object.
+
+# Populating data for the unbound column
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Unbound-Column.md:56]: provided the heading

File: MAUI/DataGrid/Unbound-Column.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            this.datagrid.Columns.Add(DiscountColumn);
+            
+
+{% endhighlight %}
+{% endtabs %}
+
+
+<img alt="" src="Images\unbound-column\maui-datagrid-sample-example.png" width="404" Height = "396" />
+
+N> It is mandatory to specify the GridColumn.MappingName for SfDataGrid.GridUnboundColumn with some name to identify the column. It is not necessary to define name of the field in the data object.
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Unbound-Column.md:54]: changed the name

File: MAUI/DataGrid/Unbound-Column.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+                Format = "C"
+            };
+            this.datagrid.Columns.Add(DiscountColumn);
+            
+
+{% endhighlight %}
+{% endtabs %}
+
+
+<img alt="" src="Images\unbound-column\maui-datagrid-sample-example.png" width="404" Height = "396" />
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Unbound-Column.md:52]: removed the height

File: MAUI/DataGrid/Unbound-Column.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+title: Unbound Columns in MAUI DataGrid control | Syncfusion
+description: Learn here all about unbound columns support in Syncfusion MAUI DataGrid (SfDataGrid) control and more about it.
+platform: MAUI
+control: SfDataGrid
+documentation: UG
+---
+
+# Unbound column in MAUI DataGrid (SfDataGrid)
+
+The data grid allows adding additional columns that are not bound with data objects from the underlying data source.The unbound column can be added using the [SfDataGrid.DataGridUnboundColumn] class.
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Unbound-Column.md:12]: Added the brackets

File: MAUI/DataGrid/Unbound-Column.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+{% endhighlight %}
+{% endtabs %}
+
+
+<img alt="" src="Images\unbound-column\maui-datagrid-sample-example.png" width="404" Height = "396" />
+
+N> It is mandatory to specify the GridColumn.MappingName for SfDataGrid.GridUnboundColumn with some name to identify the column. It is not necessary to define name of the field in the data object.
+
+# Populating data for the unbound column
+Data for the unbound column can be configured by setting the `Expression` or `Format` property..
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Unbound-Column.md:57]: removed the format property

File: MAUI/DataGrid/Unbound-Column.md [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+
+The DataGridUnboundColumnEventArgs provides the following properties:
+
+***Column:*** Gets DataGridColumn of the cell that triggers this event.
+***OriginalSender:*** Gets the data grid raising event.
+***Record:*** Gets the underlying row data
+***UnboundAction:*** Defines the action for triggering event.
+***Value:*** Gets or Sets the value for DataGridUnboundCOlumn cell based on unboundAction.
+
+# Populate the data for the unbound column by handling
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Unbound-Column.md:180]: Removed the heading

File: MAUI/DataGrid/overview.md [hunk 8]
```diff
@@ -51,6 +51,8 @@ The .NET MAUI DataGrid control is used to display and manipulate data in a tabul
 
 * **Exporting** - Support to export the data to Excel and PDF.
 
- **Unbound Row** - It allows adding rows at the top and bottom of the DataGrid which are not bound with data objects from the underlying data source.
+* **Unbound column** - The data grid allows adding additional columns that are not bound with data objects from the underlying data source.
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/overview.md:54]: changed the content.

File: MAUI/DataGrid/sorting.md [hunk 9]
```diff
@@ -245,6 +245,22 @@ public class CustomSortComparer : IComparer<object>, ISortDirection
...
+
+{% highlight xaml %}
+<syncfusion:SfDataGrid.DefaultStyle>
+    <syncfusion:DataGridStyle SortIconColor="DodgerBlue" />
+</syncfusion:SfDataGrid.DefaultStyle>
+{% endhighlight %}
+{% endtabs %}
+
+![DataGrid with sort template selector]
+<img src="Images\sorting\maui-Sort-Icon-Color.png" width="404">
```
[the remark this answers is not in the document]
Comment (AbhishekSF4201) [reply MAUI/DataGrid/sorting.md:261]: Provided image name as per the guidelines

File: MAUI/DataGrid/sorting.md [hunk 10]
```diff
@@ -245,6 +245,34 @@ public class CustomSortComparer : IComparer<object>, ISortDirection
...
+                              HeaderText="Customer ID" />
+                    <syncfusion:DataGridTextColumn MappingName="ShipCountry "
+                              HeaderText="Ship Country " />
+            </syncfusion:SfDataGrid.Columns>
+</syncfusion:SfDataGrid>
+{% endhighlight %}
+{% endtabs %}
+
+
+<img alt="DataGrid with sort template selector" src="Images\sorting\maui-datagrid-sort-icon-color.png" width="404">
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/sorting.md:273]: Alt text should be "Sort icon customization in MAUI DataGrid"

File: MAUI/DataGrid/sorting.md [hunk 11]
```diff
@@ -245,6 +245,34 @@ public class CustomSortComparer : IComparer<object>, ISortDirection
...
+The default sort icon color can be customized by setting the `DataGridStyle.SortIconColor` property.
+{% tabs %}
+
+{% highlight xaml %}
+<syncfusion:SfDataGrid x:Name="sfDataGrid"
+                        ItemsSource="{Binding OrderInfoCollection}" >
+        <syncfusion:SfDataGrid.DefaultStyle>
+                <syncfusion:DataGridStyle SortIconColor="DodgerBlue" />
+        </syncfusion:SfDataGrid.DefaultStyle>
+            <syncfusion:SfDataGrid.Columns>
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/sorting.md:259]: Columns tag should start with DefaultStyle tag's start position.

File: MAUI/DataGrid/Unbound-Column.md [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+***Value:*** Gets or Sets the value for DataGridUnboundCOlumn cell based on unboundAction.
+
+Populate the data for the unbound column by handling `QueryUnboundColumnValue` event which allows customizing the value of the `DataGridUnboundColumn`. `DataGridUnboundColumnEventArgs` exposes `Value` property, by which you can set the value for the grid cells of the unbound column based on the `UnboundAction`.
+
+Refer to the following code example in which data for the unbound column is populated by handling the `QueryUnboundColumnValue` event:
+
+{% tabs %}
+{% highlight C# %}
+
+    datagrid.QueryUnboundColumnValue += DataGrid_QueryUnboundColumnValue;
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Unbound-Column.md:171]: Start without front spacing.

File: MAUI/DataGrid/Unbound-Column.md [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+    private void DataGrid_QueryUnboundColumnValue(object? sender, DataGridUnboundColumnEventArgs e)
+    {
+        if (e.UnboundAction == DataGridUnboundActions.QueryData)
+        {
+            var Price1 = Convert.ToInt16(e.Record.GetType().GetProperty("Price1").GetValue(e.Record));
+            var Price2 = Convert.ToInt16(e.Record.GetType().GetProperty("Price2").GetValue(e.Record));
+            var amount = Price1 - Price2;
+            e.Value = amount;
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Unbound-Column.md:179]: Why don't you directly set e.Value = Prince1 - Price2?

File: MAUI/DataGrid/Unbound-Column.md [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+                Format = "C"
+            };
+            this.datagrid.Columns.Add(DiscountColumn);
+            
+
+{% endhighlight %}
+{% endtabs %}
+
+
+<img alt="Unbound-column" src="Images\unbound-column\maui-datagrid-unbound-column.png" width="404" />
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Unbound-Column.md:52]: Alt text should be "Unbound column in MAUI DataGrid".

File: MAUI/DataGrid/Unbound-Column.md [hunk 15]
```diff
@@ -0,0 +1,184 @@
...
+                        Expression="Price1+Price2"
+                        Format="C" />
+    </syncfusion:SfDataGrid.Columns>
+</syncfusion:SfDataGrid>
+
+{% endhighlight %}
+
+{% highlight c# %}
+
+            SfDataGrid dataGrid = new SfDataGrid();
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/Unbound-Column.md:35]: Start codes without front spacing.

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]

File: MAUI/DataGrid/sorting.md (hunk 10 above)
Comment (AbhishekSF4201) [reply MAUI/DataGrid/sorting.md:273]: changed

File: MAUI/DataGrid/sorting.md (hunk 11 above)
Comment (AbhishekSF4201) [reply MAUI/DataGrid/sorting.md:259]: changed

File: MAUI/DataGrid/Unbound-Column.md (hunk 12 above)
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Unbound-Column.md:171]: addressed the code

File: MAUI/DataGrid/Unbound-Column.md (hunk 14 above)
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Unbound-Column.md:52]: changed the alt

File: MAUI/DataGrid/Unbound-Column.md (hunk 15 above)
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Unbound-Column.md:35]: refactor the code

File: MAUI/DataGrid/Unbound-Column.md (hunk 13 above)
Comment (AbhishekSF4201) [reply MAUI/DataGrid/Unbound-Column.md:179]: changed

**Review by ashok-kuvaraja — APPROVED** [review]
The content changes are fine.

**Review by ashok-kuvaraja — APPROVED** [review]
The changes are fine.

**Pull request merged**